### PR TITLE
Support curation npm tree calc by package-lock only

### PR DESCRIPTION
**Pull request by asafambar**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/commands/audit/sca/npm/npm.go [hunk 1]
```diff
@@ -28,10 +29,18 @@ func BuildDependencyTree(npmArgs []string) (dependencyTrees []*xrayUtils.GraphNo
...
+	treeDepsParam := biutils.NpmTreeDepListParam{
+		Args: addIgnoreScriptsFlag([]string{}),
+	}
+	if params != nil {
+		treeDepsParam = biutils.NpmTreeDepListParam{
+			Args:                 addIgnoreScriptsFlag(params.Args()),
+			IgnoreNodeModules:    params.NpmIgnoreNodeModules(),
+			OverWritePackageLock: params.NpmOverwritePackageLock(),
+		}
+	}
```
Comment (yahavi) [inline xray/commands/audit/sca/npm/npm.go:41]: Let's export this logic to a new private method "createTreeDepsParam".
We must keep the methods in the code as short as possible.

File: xray/utils/auditbasicparams.go [hunk 2]
```diff
@@ -18,6 +18,8 @@ type AuditBasicParams struct {
 	args                    []string
 	depsRepo                string
 	ignoreConfigFile        bool
+	npmIgnoreNodeModules    bool
+	npmOverWritePackageLock bool
```
Comment (yahavi) [inline xray/utils/auditbasicparams.go:22]: I'm not sure it is a good idea to have npm specific arguments in a generic audit struct.
Can we use inheritance? Something like:
```go
type AuditNpmParams struct {
	AuditBasicParams
	npmIgnoreNodeModules    bool
	npmOverWritePackageLock bool
}
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: xray/utils/auditbasicparams.go (hunk 2 above)
Comment (asafambar) [reply xray/utils/auditbasicparams.go:22]: we already have specific arguments for python in the "generic" audit struct (was there before I did my changes)
I thought that once I will support more pkg type, I will do something more generic

Comment (asafambar) [reply xray/utils/auditbasicparams.go]: I've pushed changes with refactor, please cheeck it out 🙏

File: xray/commands/audit/sca/npm/npm.go [hunk 3]
```diff
@@ -28,10 +28,14 @@ func BuildDependencyTree(npmArgs []string) (dependencyTrees []*xrayUtils.GraphNo
...
 		return
 	}
-	npmArgs = addIgnoreScriptsFlag(npmArgs)
-
+	treeDepsParam := biutils.NpmTreeDepListParam{
+		Args: addIgnoreScriptsFlag([]string{}),
+	}
+	if params != nil {
+		treeDepsParam = createTreeDepsParam(params)
+	}
```
Comment (yahavi) [inline xray/commands/audit/sca/npm/npm.go:36]: Let's create NpmTreeDepListParam only in createTreeDepsParam:
```suggestion
	treeDepsParam := createTreeDepsParam(params)
```

File: xray/commands/curation/curationaudit.go [hunk 4]
```diff
@@ -192,8 +192,18 @@ func (ca *CurationAuditCommand) doCurateAudit(results map[string][]*PackageStatu
 	return nil
 }
 
+func (ca *CurationAuditCommand) getAuditParamsByTech(tech coreutils.Technology) utils.AuditParams {
+	if tech == coreutils.Npm {
+		return utils.AuditNpmParams{
+			AuditParams: ca.AuditParams}.
```
Comment (yahavi) [inline xray/commands/curation/curationaudit.go:198]:
```suggestion
		return utils.AuditNpmParams{AuditParams: ca.AuditParams}.
```

File: xray/commands/audit/sca/npm/npm.go [hunk 5]
```diff
@@ -46,6 +50,17 @@ func BuildDependencyTree(npmArgs []string) (dependencyTrees []*xrayUtils.GraphNo
 	return
 }
 
+func createTreeDepsParam(params utils.AuditParams) biutils.NpmTreeDepListParam {
+	npmTreeDepParam := biutils.NpmTreeDepListParam{
```
Comment (yahavi) [inline xray/commands/audit/sca/npm/npm.go:54]:
```suggestion
        if params == nil {
        	return biutils.NpmTreeDepListParam{Args: addIgnoreScriptsFlag([]string{})}
        }
	npmTreeDepParam := biutils.NpmTreeDepListParam{
```

File: xray/utils/packageManagerAuditParams.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package utils
```
Comment (yahavi) [inline xray/utils/packageManagerAuditParams.go:1]: Let's rename the file to auditnpmparams.go

File: xray/utils/packageManagerAuditParams.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+type AuditNpmParams struct {
+	AuditParams
+	npmIgnoreNodeModules    bool
+	npmOverWritePackageLock bool
```
Comment (yahavi) [inline xray/utils/packageManagerAuditParams.go:6]:
```suggestion
	npmOverwritePackageLock bool
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: xray/utils/packageManagerAuditParams.go [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+func (abp AuditNpmParams) SetNpmIgnoreNodeModules(ignoreNpmNodeModules bool) AuditNpmParams {
```
Comment (yahavi) [inline xray/utils/packageManagerAuditParams.go:9]: Let's do it in all methods in this file
```suggestion
func (anp AuditNpmParams) SetNpmIgnoreNodeModules(ignoreNpmNodeModules bool) AuditNpmParams {
```

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**